src/symbol.cr

```diff
@@ -78,6 +78,7 @@ struct Symbol
   # :nodoc:
   # Determines if a string needs to be quoted to be used for an external
   # parameter name or a named argument's key.
+  @[Deprecated("Use `.quite_for_named_argument` instead")]
```

Comment:
```suggestion
  @[Deprecated("Use `.quote_for_named_argument` instead")]
```

Also a deprecated method should never be used to implement a non-deprecated method.

Reply:
Right. The idea was to make it a private method after deprecation. But I suppose we'd have to make a new private method and delegate `needs_quotes_for_named_argument?` to that. Which is all more hassle than just keeping this method around without deprecation. It's nodoc anyway, so that's fine.